MQ Exception when running stop() on StandardIntegrationFlow

When running `.stop()` for the following code;

```
public StandardIntegrationFlow queueReader(ConnectionFactory connectionFactory, String queueName) {
	return IntegrationFlows.from(
		Jms
			.inboundAdapter(connectionFactory)
			.destination(queueName)
		, e -> e.poller(
			Pollers
				.fixedDelay(1000)
				.errorChannel(CHANNEL_ERROR)
				.maxMessagesPerPoll(1)
		)
	)
	.channel(CHANNEL_BACKUP)
	.get();
}
```
I get an error and the flow is redirected to CHANNEL_ERROR as expected. In the error handler `payload.getFailedMessage()` is `null` which is (kind-of) expected as the queue was empty. Restarting with `.start()` is working OK and messages are processed.

This is possibly a JMS or IBM MQ issue, but is Spring Integration properly interacting with JMS or MQ? (I understand that this may not be an S.I. issue, but it would help to know how S.I. is stopping in order to pass on the issue to JMS or IBM).

Possibly, the `stop()` is interrupting the lower level `JMSTemplate.doReceive()` thus causing the exception. Maybe, the `stop()` should just cause the next poll to ignore doReceive(). I know that from working with MQ classes that a `get` can (depending on the API call and/or MQ GET options) wait for a period before returning (similar to a Java `BlockingQueue`) and maybe it is this that is blocking. When I interrupt (the MQ class) I _can_ get 2195.

The trace is as follows;
```
Caused by: org.springframework.jms.UncategorizedJmsException: Uncategorized exception occurred during JMS processing; nested exception is com.ibm.msg.client.jms.DetailedJMSException: JMSWMQ2002: Failed to get a message from destination 'QA.UAT.GLOBIT.TO.DRS.OPTIONS'.
IBM MQ classes for JMS attempted to perform an MQGET; however IBM MQ reported an error.
Use the linked exception to determine the cause of this error.; nested exception is com.ibm.mq.MQException: JMSCMQ0001: IBM MQ call failed with compcode '2' ('MQCC_FAILED') reason '2195' ('MQRC_UNEXPECTED_ERROR').
	at org.springframework.jms.support.JmsUtils.convertJmsAccessException(JmsUtils.java:311)
	at org.springframework.jms.support.JmsAccessor.convertJmsAccessException(JmsAccessor.java:185)
	at org.springframework.jms.core.JmsTemplate.execute(JmsTemplate.java:507)
	at org.springframework.jms.core.JmsTemplate.receiveSelected(JmsTemplate.java:747)
	at org.springframework.integration.jms.JmsDestinationPollingSource.doReceiveJmsMessage(JmsDestinationPollingSource.java:157)
	at org.springframework.integration.jms.JmsDestinationPollingSource.doReceive(JmsDestinationPollingSource.java:126)
	at org.springframework.integration.endpoint.AbstractMessageSource.receive(AbstractMessageSource.java:167)
	at org.springframework.integration.endpoint.SourcePollingChannelAdapter.receiveMessage(SourcePollingChannelAdapter.java:250)
	at org.springframework.integration.endpoint.AbstractPollingEndpoint.doPoll(AbstractPollingEndpoint.java:359)
	at org.springframework.integration.endpoint.AbstractPollingEndpoint.pollForMessage(AbstractPollingEndpoint.java:328)
	... 14 more
Caused by: com.ibm.msg.client.jms.DetailedJMSException: JMSWMQ2002: Failed to get a message from destination 'QA.UAT.GLOBIT.TO.DRS.OPTIONS'.
IBM MQ classes for JMS attempted to perform an MQGET; however IBM MQ reported an error.
Use the linked exception to determine the cause of this error.
	at com.ibm.msg.client.wmq.common.internal.Reason.reasonToException(Reason.java:595)
	at com.ibm.msg.client.wmq.common.internal.Reason.createException(Reason.java:215)
	at com.ibm.msg.client.wmq.internal.WMQMessageConsumer.checkJmqiCallSuccess(WMQMessageConsumer.java:222)
	at com.ibm.msg.client.wmq.internal.WMQMessageConsumer.checkJmqiCallSuccess(WMQMessageConsumer.java:156)
	at com.ibm.msg.client.wmq.internal.WMQConsumerShadow.getMsg(WMQConsumerShadow.java:1858)
	at com.ibm.msg.client.wmq.internal.WMQSyncConsumerShadow.receiveInternal(WMQSyncConsumerShadow.java:228)
	at com.ibm.msg.client.wmq.internal.WMQConsumerShadow.receive(WMQConsumerShadow.java:1461)
	at com.ibm.msg.client.wmq.internal.WMQMessageConsumer.receive(WMQMessageConsumer.java:674)
	at com.ibm.msg.client.jms.internal.JmsMessageConsumerImpl.receiveInboundMessage(JmsMessageConsumerImpl.java:1051)
	at com.ibm.msg.client.jms.internal.JmsMessageConsumerImpl.receive(JmsMessageConsumerImpl.java:667)
	at com.ibm.mq.jms.MQMessageConsumer.receive(MQMessageConsumer.java:209)
	at org.springframework.jms.support.destination.JmsDestinationAccessor.receiveFromConsumer(JmsDestinationAccessor.java:132)
	at org.springframework.jms.core.JmsTemplate.doReceive(JmsTemplate.java:788)
	at org.springframework.jms.core.JmsTemplate.doReceive(JmsTemplate.java:765)
	at org.springframework.jms.core.JmsTemplate.lambda$receiveSelected$9(JmsTemplate.java:749)
	at org.springframework.jms.core.JmsTemplate.execute(JmsTemplate.java:504)
	... 21 more
Caused by: com.ibm.mq.MQException: JMSCMQ0001: IBM MQ call failed with compcode '2' ('MQCC_FAILED') reason '2195' ('MQRC_UNEXPECTED_ERROR').
	at com.ibm.msg.client.wmq.common.internal.Reason.createException(Reason.java:203)
	... 35 more
```

--- a/src/reference/asciidoc/jms.adoc
+++ b/src/reference/asciidoc/jms.adoc
@@ -87,6 +87,17 @@ If you prefer to have the raw JMS message as the Spring Integration message's pa
 ====
 
 Starting with version 5.0.8, a default value of the `receive-timeout` is `-1` (no wait) for the `org.springframework.jms.connection.CachingConnectionFactory` and `cacheConsumers`, otherwise it is 1 second.
+The JMS Inbound Channel Adapter crates a `DynamicJmsTemplate` based on the provided `ConnectionFactory` and options.
+If an external `JmsTemplate` is required (e.g. in Spring Boot environment), or `ConnectionFactory` is not caching, or no `cacheConsumers`, it is recommended to set `jmsTemplate.receiveTimeout(-1)` if a non-blocking consumption is expected:
+
+====
+[source,java]
+----
+Jms.inboundAdapter(connectionFactory)
+        .destination(queueName)
+        .configureJmsTemplate(template -> template.receiveTimeout(-1))
+----
+====
 
 [[jms-ib-transactions]]
 ==== Transactions

